test/upgrade/types.go

```diff
@@ -91,6 +92,17 @@ type StopEvent struct {
 	name     string
 }
 
+// WaitOnStopEventConfiguration holds a values to be used be WaitForStopEvent
+// function. Handler will be called when StopEvent is sent. OnWait will be
+// invoked in a loop while waiting, and each wait act is driven by WaitTime
+// amount.
+type WaitOnStopEventConfiguration struct {
+	Name     string
+	Handler  func(event StopEvent) interface{}
```

Comment:
Would be great to name this `OnStop` . This aligns nicely with `OnWait` and is also self-explanatory. `Handler` is pretty generic.

Reply:
I was wondering why the Handler function returns `interface{}` . AFAIK,  the user can never use the return value and it is only printed in debug logs. However, the user can already log whatever they want in the Handler function. So, could we make the return type more explicit? (e.g. int)

Reply:
I removed the return value